fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/mixin/structure/MixinChunkGeneratorType.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.structure;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.world.gen.chunk.ChunkGeneratorType;
+
+import net.fabricmc.fabric.impl.structure.FabricStructuresImpl;
+
+@Mixin(ChunkGeneratorType.class)
+public class MixinChunkGeneratorType {
+	@Inject(method = "method_30641", at = @At("RETURN"))
+	private static void onCreateCavesType(CallbackInfoReturnable<ChunkGeneratorType> cir) {
+		FabricStructuresImpl.defaultStructuresConfigs.add(cir.getReturnValue().getConfig());
```

Comment:
I remember modmuss had to add something to biome api since nether chunk generator was initialized before mod init. Maybe validate this isn't called too early before chunk generator type is created?

Reply:
No, it's safe in this case. The `DEFAULT_STRUCTURES_CONFIGS` set stores the set of configs which copied from `StructureFeature.DEFAULT_STRUCTURES`. This is so that when a mod registers a structure after the `StructuresConfig` has been created, the `register` method loops over all existing configs and adds the structure to it. If the structure is registered before this `StructureConfig` is created, then since it's added to `StructureFeature.DEFAULT_STRUCTURES` the `StructureConfig` will copy the structure into its list anyway.